### PR TITLE
Replace botched Tyrian 2000 setup fix with a (hopefully) proper one

### PR DESCRIPTION
**Pull request by FeralChild64**
# Description

This replaces the old (improper) fix for the _Tyrian 2000_ hardware setup tool with a port 0x60 timing fix.

## Related issues

Fixes https://github.com/dosbox-staging/dosbox-staging/issues/3691
https://github.com/dosbox-staging/dosbox-staging/issues/3632

# Manual testing

- checked that _Tyrian 2000_ hardware setup tool has usable menu
- played _Gods_ for a while, checked that pressed keys no longer appear to get stuck
- checked that it is still possible to type player name in _Ultima VIII: Pagan_
- tested _Windows 3.11 for Workgroups_ with standard PS/2 mouse driver, VirtualBox mouse driver, A4tech 3DMouse driver
- played some games for a while - Wolfenstein 3D, DOOM, Duke Nukem 3D, Grand Theft Auto

# Checklist

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: src/hardware/input/intel8042.cpp [hunk 1]
```diff
@@ -1024,7 +1018,10 @@ static uint8_t read_data_port(io_port_t, io_width_t) // port 0x60
 	is_data_from_aux = false;
 	is_data_from_kbd = false;
 
-	maybe_transfer_buffer();
+	// Enforce the simulated data transfer delay, as some software
+	// (Tyrian 200 setup) reads the port without waiting for the
```
Comment (dreamer) [inline src/hardware/input/intel8042.cpp:1022]: Nitpick: it's Tyrian 200**0** ;)

Comment (FeralChild64) [reply src/hardware/input/intel8042.cpp]: Fixed.

**Comment by johnnovak** [comment]
I'm glad it looks like it's a simple fix in the end, @FeralChild64 🎉 I will test this with a wide variety of at least 20-30 games and report back.

**Comment by johnnovak** [comment]
Tested these games; keyboard input is working fine in all of them:

- 2400 A.D.
- Another World
- Dungeon Master
- Earthworm Jim
- Flashback
- Gauntlet
- Gods
- Jazz Jackrabbit Holiday Hare
- Lost Vikings, The
- Magic Pockets
- Mario and Luigi
- Might and Magic – Book Two
- Myst (Win 3.11)
- Neuromancer
- Paku Paku
- Pinball Dreams
- Pinball Illusions
- Pool of Radiance
- Prince of Persia
- Quest for Glory I (EGA)
- Rick Dangerous 2
- Speedball 2
- System Shock (CD Version)
- Times of Lore
- Tomb Raider Gold
- Total Eclipse
- Tricky Quiky Games
- Ultima IV
- Ultima V
- Ultima VI
- Ultimate Underworld
- Wasteland
- Windwalker
- Wizardry I (CGA)
- Wizardry VI - Bane of the Cosmic Forge
- X-COM – Terror From the Deep
- Zak McKracken Enhanced (Tandy)
- Zool

**Review by johnnovak — APPROVED** [review]
Couldn't find any keyboard handling regressions in the 38 games I tested, so this seems promising @FeralChild64. It definitely looks like a _more proper fix_ than your previous attempt, and likely closer to how the hardware behaves. 😆

**Comment by gobalius** [comment]
@johnnovak I'm just curious... I have now tested Gods with dosbox-staging-windows-x64-0.82.0-alpha-10320 and for me in the game's main menu I still can't select ENTER PASSWORD because the cursor goes so fast it either skips all the way to the top or the bottom option... were you also focused on what is going on in the menus?

**Comment by johnnovak** [comment]
> @johnnovak I'm just curious... I have now tested Gods with dosbox-staging-windows-x64-0.82.0-alpha-10320 and for me in the game's main menu I still can't select ENTER PASSWORD because the cursor goes so fast it either skips all the way to the top or the bottom option... were you also focused on what is going on in the menus?

Yeah that's normal. It works like that on hardware too. Just touch the cursor keys very briefly in the menu.

**Comment by johnnovak** [comment]
@FeralChild64 Can we backport only this fix without the proper Ultima VIII – Pagan fix? I assume we can, and it would be worthwhile, right?

I'm inclined to release 0.81.2 just with this fix. Thoughts @weirddan455 @interloper98 @kklobe @shermp ?

**Comment by FeralChild64** [comment]
Sure. In fact, that’s why I have put them in a separate pull requests :)

**Comment by weirddan455** [comment]
I believe Tyrian 2000 worked fine in 0.81.1.  @FeralChild64 's original fix for Tyrian 2000 was the main reason to release 0.81.1 in the first place as it had a regression in 0.81.0.

The Tyrian 2000 fix caused keyboard issues in Gods and MegaZeux and were solved in this PR.  Those are the only 2 games I know of with problems, but there may be other.

I'm good with releasing an 0.81.2 assuming this patch applies cleanly on top of 0.81.1.  I've lost track of other bug fixes we've made in `main` since the last release but this is probably the one that will affect the most users.

**Pull request merged**